build_and_push.sh

```diff
@@ -54,9 +54,16 @@ function build_and_push_manifest() {
     done
 }
 
+function read_config() {
+    CONFIG_KEY="$1"
+    CONFIG_FILE="${2:-build-versions.yaml}"
 
-for spackver in "v0.20.3" "v0.21.0" ; do
-    for baseimg in docker.io/ubuntu:22.04 ; do
+    yq ".${CONFIG_KEY}[]" ${CONFIG_FILE}
```

Comment:
I am almost sure that it should be `yq -r`, otherwise the values will have quotes around them, which is not what we want. 
Did you try to build the images with the PR, I did not, but I would assume that it fails due to the quotes. For comparison
```bash
anfink@anfink-laptop /home/tmp $ for x in $(yq ".spackver[]" test.yaml) ; do echo "x=$x" ; done
x="v0.20.3"
x="v0.21.0"
```
vs
```bash
anfink@anfink-laptop /home/tmp $ for x in $(yq -r ".spackver[]" test.yaml) ; do echo "x=$x" ; done
x=v0.20.3
x=v0.21.0
```